code/modules/mob/living/silicon/robot/robot_model.dm

```diff
@@ -578,6 +577,11 @@
 	if(lube)
 		lube.reagents.add_reagent(/datum/reagent/lube, 2 * coeff)
 
+	var/obj/item/soap/cyborg/soap = locate(/obj/item/soap/cyborg) in basic_modules
+	if(soap)
+		if(soap.uses < 300)
+			soap.uses += 3 * coeff
```

Comment:
You shouldn't hardcode the size of the soap like this, cause what if someone changes it in future right? then we're boned.

You can use the initial() proc to access a thing's var as it was initially, I think that'll be useful for both the limit check and the addition <https://www.byond.com/docs/ref/#/proc/initial>